common/src/KokkosFFT_layouts.hpp

```diff
@@ -30,10 +30,8 @@ auto get_extents(const InViewType& in, const OutViewType& out,
                      "input axes are not valid for the view");
 
   constexpr std::size_t rank = InViewType::rank;
-  [[maybe_unused]] int inner_most_axis =
-      std::is_same_v<array_layout_type, typename Kokkos::LayoutLeft>
-          ? 0
-          : (rank - 1);
+  int inner_most_axis =
+      std::is_same_v<array_layout_type, Kokkos::LayoutLeft> ? 0 : (rank - 1);
```

Comment:
This variable `inner_most_axis` is actually unused for `C2C` transform which is identified at compile time. 
I thought in this case compilers regard this variable unused. 
However, I found no compiler errors

Reply:
Could be because there is no `if constexpr` ?

Reply:
You are right. lines 64 and 73 must be `if constexpr`.

Reply:
If you are fine, I will merge this and fix `if constexpr` in another PR.

Reply:
Yes, you might need `[[maybe_unused]]` then.

Reply:
That is exactly what I had in mind